### ISSUE TITLE
Idea: Delete Waypont screen entirely

### ISSUE THREAD

**Issue by sonora** [osmandapp/Osmand#2158]
I guess we could now delete the Waypoints screen entirely, and move its functionality as we have developed it now (switch/swipe destinations, configure POI / Favorites / Warnings display, etc.) onto the route planning dialog, where we combine it with the functionality we already have there  (Add destinations, Show route details), i.e. essentially combine the 2 screens onto 1 ?

**Comment by raindropsfromsky** [comment]
Some additional points:
 1. If there are no waypoints, the ***Waypoints*** screen shows the start point. But if we add 1+ waypoints, the screen stops showing the start point. It may appear after some time.
 2. The "Door-to-door sorting" does not mean anything. It should be renamed "optimize route"
 3. The "Door-to-door sorting" function does not belong to the start point. It should be launched from a button in the header/footer of the dialog.
 4. In the ***Waypoints*** GUI, the points have a second line with tiny text, identifying them as start point, x. destination, destination, etc. These labels are NOT needed: The list is always in top-down order; so the first point is always the start point, and the last point is always the destination. So the labels state the obvious.
 5. If we activate the ***POI*** option, it shows a huge list of POIs. Here, the obvious intent is to select one of those (well, max 2-3). So we have to a large number of unwanted POIs by tapping on the X buttons. Instead, provide check boxes against all POIs, so that we can select just the few points we want; and let go of the rest.
 6. We cannot move the first point like the rest. Please allow this. (In other words, the first point should be just like the rest)

**Comment by raindropsfromsky** [comment]
10. When POIs or Favorites are added as waypoints (see points 5/6 in the previous post), the screen should show their original icons, not as yellow triangular flags.

**Comment by vshcherb** [comment]
We don't want to delete waypoint screen we want to delete Waypoints item from Drawer.
It is 1 screen anyway. We plan to delete in branch first to see how it goes.
It is work in progress so I will close this issue

**Comment by sonora** [comment]
Victor, afraid Narayan watered my original issue and slightly shifted the scope: I seriously suggest to combine waypoint screen with what we now have as "Route planning screen",we do not need 2 seperate screens anymore! Think about it: Setting targets and shifting, deleting, editing them all belongs (and fits, the way we have it now!) on one screen. (Then we can of course also delete the ref from the drawer, as "Directions" would (and already does) point right to the screen I am suggesting.

**Comment by vshcherb** [comment]
I will leave this open for now, I also had that intention to combine basically 3 screens: route settings, route overview (not a route itinerary) and route waypoints into 1. But I'm not that confident any more, waypoints quite complex and long screen (it could include list of pois or favorites if enabled), having it opened while navigation could a big shift. Though I will ask others about that idea how is it feasible. Personally after some usage that setup doesn't disturb me much (previous was irritating with jumping buttons)

**Comment by raindropsfromsky** [comment]
Hi

PFA proposed layout for a single-window Route Editor.

IMHO this design will allow a smooth intuitive workflow.
[RouteEditor.zip](https://github.com/osmandapp/Osmand/files/94145/RouteEditor.zip)

BTW also see  #2170, where I proposed to separate the route-planning from the pop-up cards that show POIs and Favorites along the route.

**Comment by sonora** [comment]
Victor, I agree, coding-wise it is more involved than make it sound  :-)

But here is the breakdown:
(1) Use the current "Waypoints" screen instead of the current "Route planning dialog"
(2) To its top row, add the profile selector
(3) In its "Destinations" list, add the single-tap actions which  open the "Spinner" actions, i.e. add/change destinations (Popup Menu "Select on map  / Favorites / Address, etc.")
(4) Near its bottom, add the "Route details" item line
(5) At the very bottom, display the route controls (Cancel / Waypoints / Route Settings / GO)

We did a similar trick with the "Route Settings" screen. Let use do the same with the "Waypoints". screen. Of course, the screen will increase in length, and it must be scrollable. But the important portion is the top, where the destinations are shown, all other items can be accessed via scrolling, or may not be needed for the majority of users anyway.

(Yes, I know it is WIP. But I wanted to bring this up while we still work on it, so we do not use dev hours, then have to re-visit...:-) ) - Best, Hardy

**Comment by vshcherb** [comment]
Good that you brought it now, but I'm afraid the design is more or less finished for next release, though implementation is not. I see your point, basically put destinations on top and scrollable content, I'm not sure I exactly like it, but it is worth to draw it in whatever possible, cause it is hard to undertsand the text.

**Comment by sonora** [comment]
Here is my mockup: Everything but the bottom row (the route controls) should be scrollable.
![mockup](https://cloud.githubusercontent.com/assets/845735/12404300/94a6e95a-be3a-11e5-8e38-6ed1eee3b291.png)
)

**Comment by vshcherb** [comment]
These kind of "Tabs" on the bottom (Flag 1 screen, Settings 2nd screen) is a bad UI pattern. We used to have it previous release, but there is no such controls. If we want to make 2 tabs: Waypoints and Settings they should be on the top, highlight will make it bad and that's what we try to avoid in the new UI.

**Comment by raindropsfromsky** [comment]
I realized that two buttons were missing in my proposed layout:
 1. Add current location
 2. Switch to the tour itinerary ("Route details" screen)

The updated layout is given below:
![route editor](https://cloud.githubusercontent.com/assets/9047168/12410358/392c4cd6-be9a-11e5-895d-fc30aa4ab200.png)

**Comment by raindropsfromsky** [comment]
Instead of providing five separate buttons to add five types of points, we can have a single ***Add*** drop-down menu.

**Comment by RicoZ-OSM** [comment]
points and waypoints sure need a cleanup of UI.

After switching from 2.1.1 to 2.2.4 I am completely and utterly lost:
* when I quit navigation - is there any quick way to delete all "flags" (dest, waypoint..)?
* the "delete" is almost never displayed when I crave for it
* can I insert waypoints once start and endpoint have been specified?

**Comment by raindropsfromsky** [comment]
All these points are taken care in the GUI I suggested.

And the best part is, it is just one window. No need to switch between
multiple windows...

**Comment by raindropsfromsky** [comment]
Victor,

We should have a lively debate about the relative merits of the existing design vs the two designs suggested by Sonora and me. I feel you still have some reservation about merging these various screens...

**Comment by therob84** [comment]
I really love the [2nd proposal](https://github.com/osmandapp/Osmand/issues/2158#issuecomment-172741364) of @raindropsfromsk.
I went from v2.1.1@Android 4.2 to v2.2.4@Android 5.2 ......... and really needed hours to set-up a route with 3 waypoints in the correct order. I gave up! For v2.1.1 I need 2 minutes for this!
I was really frustrated!

The worst point is that I was not able to find the manual sorting of the waypoints at all. 
I could simply not find this option in v2.2.4! Am I too stupid to just switch the order of two waypoints, or is the waypoint/routing GUI in the current version really messed up?

In general:
After playing inside a car for 3 hours with the v2.2.4 I was really disappointed and made a list of all together 16 points, what I really hate in this version and should be modified or added (most things in comparison to the much better version 2.1.1. I will post in the next days either in one post with all complains or separate it into different ones.)
But as the proposals above exactly did adress some of my problems I strongly vote to implement such kind of re-worked routing-GUI !! More details later!
